homeassistant/components/broadlink/switch.py

```diff
@@ -97,7 +98,7 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     mac_addr = binascii.unhexlify(config.get(CONF_MAC).encode().replace(b":", b""))
     switch_type = config.get(CONF_TYPE)
     retry_times = config.get(CONF_RETRY)
-    devtype = 0x5F36 if switch_type == "rm_mini3_5f36" else 0x272A
+    devtype = 0x5F36 if switch_type in ["rm_mini3_5f36", "rm4"] else 0x272A
```

Comment:
See? This design does not scale at all. We shouldn't be using strings for type. We must implement broadlink.discover().
Should I start doing it now? I have some time to spare, just point me the direction.

Reply:
@Danielhiversen I'm going to learn about config entries and see if we can do a discover() there. So we start from there, and forward the devices to the platforms. But I need you to agree with these changes, because they are going to cost me a lot of work.

This will make this integration progress on the quality scale. And we’re going to make it more user friendly. Why torture the user if we can just ask the device.

I'm gonna start doing it right now. Do you want me to open an issue so we can discuss the idea first?

Reply:
I think it is a good idea 👍